crates/ra_parser/src/grammar/items/traits.rs

```diff
@@ -5,19 +5,29 @@ use super::*;
 // test trait_item
 // trait T<U>: Hash + Clone where U: Copy {}
 // trait X<U: Debug + Display>: Hash + Clone where U: Copy {}
+// trait Z<U> = T<U>;
+// trait Z<U> = T<U> where U: Copy;
+// trait Z<U> = where Self: T<U>;
```

Comment:
It's better to move these tests directly to the `if p.eat(T![=])` condition. The main goal is *not* to test the code, the main goal is to explain which bit of code handles what

Reply:
Ahh, smart system. I think I get it.